orc8r/lib/go/protos/magma_id.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package protos
+
+import (
+	"context"
+	"strings"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+// X_MAGMA_ID is a request/response tracing header which must be carried between RPC calls
+const X_MAGMA_ID = "x-magma-id"
+
+// AddRequestId creates and returns new outgoing GRPC CTX with specified Id's header
+// Intended for GRPC client use
+func AddRequestId(ctx context.Context, id string) context.Context {
```

Comment:
minor nit: Adding a client name apart from ID might also be useful here to aid debugging

Reply:
Id is an arbitrary string, it can start with client name if desired.